### ISSUE TITLE
Autocomplete/Search incorrect result

### ISSUE THREAD

**Issue by avnenkovskyi1** [Smile-SA/elasticsuite#1277]
### Environment
All test were did on the ElasticSuite demo store http://demo.magento-elastic-suite.io, but the same behaviour with next setup:

Magento 2.3 includes Sample Data (the issue reproduces with real products as well)
Elastisearch 6.5.4
ElasticSuite 2.7.x

### Steps to reproduce
1. Type search term (in our case it "Messe") which didn't search previously. Result not (see screenshot below).

https://i.imgur.com/DrW6HFF.jpg

2. The result of searching by "Messe" term is not found products (see screenshot below)

https://i.imgur.com/CZtxmd7.png

3. Type the full word "Messenger" for the search term, autocomplete return three results (see screenshot below)

https://i.imgur.com/fva4H9W.jpg

4. The result of searching by "Messenger" term is three products (see screenshot below)

https://i.imgur.com/zisZWBV.png

5. Type search term from point 1 (in our case it "Messe") in the search input field. Autocomplete result is three products (see screenshot below).

https://i.imgur.com/n0ixIv9.jpg

6. The result of search by "Messe" term is not found products but "Related search terms" existed (see screenshot below)

https://i.imgur.com/bqeeLjE.png

### Expected result
Autocomplete/Search result should show relevance products in the first time of searching by the term.

Additionally, sort product should be for more relevance, current result displayed more actual product in the end of list

**Comment by romainruaud** [comment]
Hello,

TL;DR version : this is working as intended, and it's done like this by design.

Detailed version : 

**The autocomplete**

For now (but we may refactor this part on a near future), autocomplete is "self-learning", according exactly to the scenario you describe.

First type, you type "mess", no results, because it's unable to expand "mess" to a previously used search term (that could be "messenger").

Once someone has already searched for "messenger", this record will be put (it's legacy Magento code here) in the search terms table, allowing to expand an user typing "mess" to "messenger" search term, and retrieving associated results inside the autocompletion bar.

**The 'partial search'**

We cannot automatically extend results from "messe" to "messenger" on the search results pages, because it could lead to completely irrelevant ones as described in #1036.

This sound relevant with your example with "messe" and "messenger" but let's imagine I have a large catalog, if I search for a "hoe", should the search engine display "shoes" ? or display "pants" when someone is searching for "pan" ? and so on ... 

** To conclude **

In autocompletion, we widen the search, because we consider **the user is typing and might not have finished writing** => we widen to all search terms beginning with the same string as the one currently being typed. So "pan" will find "pans", "pants", "panda", "pancakes", and so on...

But on the search result page, we assume **the user has finished typing** (because he did hit ENTER and submit the form). So in this case, we favor exact matches to remain the most relevant.

The problem on your "messe" => "messenger" is that the both words are too far from each other to be considered as "exact matches" by the fulltext search engine.

Regards
